src/Features/Core/Portable/CodeAnalysisSuggestions/ICodeAnalysisSuggestionsConfigService.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.CodeAnalysisSuggestions;
+
+/// <summary>
+/// Service to compute project-wide code analysis diagnostics for first party analyzers/fixes that are relevant
+/// for the user and/or the code base to improve the code quality and/or code style of the code base.
+/// </summary>
+internal interface ICodeAnalysisSuggestionsConfigService : IWorkspaceService
+{
+    /// <summary>
+    /// Compute project-wide code analysis diagnostics for first party analyzers/fixes that are relevant
```

Comment:
this seems super expensive.  isn't this like as expensive as doing compilation-end work?  how does this interplay with our options about rnning analysis only on open docs and the like?